src/boot/bootctl.c

```diff
@@ -2555,6 +2556,334 @@ static int verb_reboot_to_firmware(int argc, char *argv[], void *userdata) {
         }
 }
 
+/* This only frees some fields. See boot_entry_auto */
+static void boot_entry_auto_done(BootEntry* entry) {
+        free(entry->title);
+        free(entry->version);
+        free(entry->sort_key);
+        free(entry->machine_id);
+
+        if (entry->options) {
+                for (size_t i = 0; i < SIZE_MAX; i++)
+                {
+                        if (!entry->options[i])
+                                break;
+
+                        free(entry->options[i]);
+                }
+        }
+        free(entry->options);
```

Comment:
That looks like a reimplementation of strv_free().

Reply:
Correct I missed that.